### ISSUE TITLE
[BUG] With require.js inside electron app , does not work

### ISSUE THREAD

**Issue by mafar** [tensorflow/tfjs#5292]
Here is my scenerio:
My app is using require.js and is running inside electron.
I kept getting error and then i fugred it out.
```
          ENV.registerFlag('IS_NODE', function () {
            // ignore NODE if AMD is found
            if (typeof define === 'function' && define.amd) {
              return false;
            }
            return typeof process !== 'undefined' && typeof process.versions !== 'undefined' && typeof process.versions.node !== 'undefined';
          });
```
Please Update function ```ENV.registerFlag('IS_NODE', function ```
so that it ignores NODE if AMD is found

**Comment by pyu10055** [comment]
@gbaned can you provide a full running example to expedite the reproduction of the issue. thanks

**Comment by gbaned** [comment]
> @gbaned can you provide a full running example to expedite the reproduction of the issue. thanks

@rthadur  Can you please take a look on the above comment from @pyu10055.  Thanks!

**Comment by rthadur** [comment]
@mafar In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

**Comment by mafar** [comment]
> @mafar In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!

example will require an electron mock up with require.js inside it. Too much for me :)
But let me explain plz
When require.js is running inside electorn as in my case, 


- ```tfjs``` is loaded via require.js (AMD), i dont load it via node.js (commonJs)
- but when code reaches at point as shown above , ```IS_NODE``` becomes true since we are inside electron 
- - but from here dependencies start to fail loading becuse initialy it was loaded via amd and now it switched to ```CommonJS ```
- - Simply put, if  ```tfjs``` is laoded via AMD , it should not consider ```CommonJs``` at all to avoid dependency resolution

**Comment by mattsoulanille** [comment]
We may need to add an `IS_COMMONJS` flag to decide if we can `require()` dependencies. There are some cases where knowing we're running in node is important apart from knowing whether we can `require()` dependencies (e.g. [WASM multithread support](https://github.com/tensorflow/tfjs/blob/b8024c0def121515917a03c66c9556373bb2bbf4/tfjs-backend-wasm/src/flags_wasm.ts#L39-L44)).

The only place I see where we're checking `IS_NODE` and then calling `require()` is in [file_data_source.ts](https://github.com/tensorflow/tfjs/blob/b8024c0def121515917a03c66c9556373bb2bbf4/tfjs-data/src/sources/file_data_source.ts#L46-L50). Alternatively, we could just check for commonjs here and avoid adding the new flag.

What do you think, @pyu10055?

**Comment by mafar** [comment]
> We may need to add an `IS_COMMONJS` flag to decide if we can `require()` dependencies. There are some cases where knowing we're running in node is important apart from knowing whether we can `require()` dependencies (e.g. [WASM multithread support](https://github.com/tensorflow/tfjs/blob/b8024c0def121515917a03c66c9556373bb2bbf4/tfjs-backend-wasm/src/flags_wasm.ts#L39-L44)).
> 
> The only place I see where we're checking `IS_NODE` and then calling `require()` is in [file_data_source.ts](https://github.com/tensorflow/tfjs/blob/b8024c0def121515917a03c66c9556373bb2bbf4/tfjs-data/src/sources/file_data_source.ts#L46-L50). Alternatively, we could just check for commonjs here and avoid adding the new flag.
> 
> What do you think, @pyu10055?

I agree. This wIll be a nice solution. If you provide me with a PR, I can test it using my project and report back quick if this is working

**Comment by shmishra99** [comment]
Hi @mafar, 
Apology for the late response. Could you please let us know if your issue has been resolved or you are still looking for the solution ? I would suggest you to give a try with the latest version of `@tensorflow/tfjs@4.4.0` and check if your issue is resolving or not.
Thank You !
